### PR TITLE
Upgrade to React 16

### PR DESCRIPTION
**Pull request by Graham42**
I started with the intention of adding a feature.... and then fell down the rabbit hole of upgrading dependencies.

This upgrades to React 16 and changes out the library used for the slider component.

### CODE REVIEW

**Comment by philippotto** [comment]
Thanks for your contribution! Unfortunately, I don't have much time to review and test this PR right now, so bear with me as I have to defer this for some weeks.

**Comment by Graham42** [comment]
No problem

File: app/scripts/views/collapsable_container.js [hunk 1]
```diff
@@ -1,14 +1,15 @@
 import React, { Component } from "react";
+import PropTypes from "prop-types";
 import classNames from "classnames";
 import _ from "lodash";
 
-export default class BeatVisualization extends Component {
+export default class CollapsableContainer extends Component {
```
Comment (philippotto) [inline app/scripts/views/collapsable_container.js]: Nice catch!

**Comment by philippotto** [comment]
Thanks again for taking the time to do this!

**Comment by philippotto** [comment]
By the way, I added a .prettierrc file and executed prettier on the entire code base. This PR was a bit hard to read due to all the formatting changes. Just so you know that you should pull soon to avoid merge problems for potential future work :)

**Pull request merged**